xclim/sdba/properties.py

```diff
@@ -0,0 +1,735 @@
+      The condition for a spell is variable {op} threshold.
+    thresh: str, float
+      Threshold on which to evaluate the condition to have a spell.
+      Str with units if the method is "amount".
+      Float of the quantile if the method is "quantile".
+    stat: {'mean','max','min'}
+      Statistics to apply to the resampled input (eg. 1-31 Jan 1980) and then over all years (eg. Jan 1980-2010)
+    time_res : str
+      Time resolution.
+      Eg. If 'month', the {stat} is calculated on 12 distributions for each grid point.
+      Each distribution contains the spell lenghts in a given month for all available years.
+    Returns
+    -------
+    xr.DataArray,
+      {stat} of spell length distribution when the variable is {op} the {method} {thresh}.
+
+    Examples
+    --------
+    >>> from xclim.testing import open_dataset
+    >>> pr = open_dataset(path_to_pr_file).pr
+    >>> spell_length_distribution(da=pr, op='<',thresh ='1mm d-1', time_res='season')
+    """
+    attrs = da.attrs
+    mask = ~(da.isel(time=0).isnull()).drop_vars("time")  # mask of the ocean with NaNs
+    ops = {">": np.greater, "<": np.less, ">=": np.greater_equal, "<=": np.less_equal}
+
+    # threshold is an amount that will be converted to the right units
+    if method == "amount":
+        t = convert_units_to(thresh, da)
+    # threshold is calculated from quantile of distribution at the time_res
+    elif method == "quantile":
+        if time_res != "year":
+            da = da.groupby(f"time.{time_res}")
+        t = da.quantile(thresh, dim="time").drop_vars("quantile")
+    else:
+        raise ValueError(
+            f"{method} is not a valid method. Choose 'amount' or 'quantile'."
+        )
+
+    cond = ops[op](da, t)
+    if time_res != "year":
+        cond = cond.resample(time=res2freq[time_res])
+        # the stat is taken on each resampling (1-31 Jan 1980)
+        out = cond.map(rl.rle_statistics, dim="time", reducer=stat)
+        # then again on all years (Jan 1980-2010)
+        out = getattr(out.groupby(f"time.{time_res}"), stat)(dim="time")
+    else:
+        out = rl.rle_statistics(cond, dim="time", reducer=stat)
+    out = out.where(mask, np.nan)  # put NaNs back over the ocean
+    out.attrs.update(attrs)
+    out.attrs[
+        "long_name"
+    ] = f"{stat} of spell length when {attrs['long_name']} {op} {method} {thresh}"
+    return out
+
+
+@update_xclim_history
+@register_statistical_properties(aspect="temporal", seasonal=True, annual=False)
+def acf(da: xr.DataArray, lag: int = 1, time_res: str = "season") -> xr.DataArray:
+    r"""Autocorrelation function.
+
+    Autocorrelation with lag-{lag} over a {time_res} and averaged over all years.
+
+    Parameters
+    ----------
+    da : xr.DataArray
+      Variable on which to calculate the diagnostic.
+    lag: int
+      lag.
+    time_res : {'season', 'month'}
+      Time resolution. 'year' is not an option for this property.
```

Comment:
I'm not sure I understand why we can't compute the autocorrelation function with `time_res='year'`?

Reply:
I was following the Measures and Indices table (http://www.value-cost.eu/validationportal/app/#!indices). The column annual is false for ACX.bias, I couldn't find why it was that way...